pagopa/io-app-design-system
# [IOAPPX-330] Allow font scaling to all the typographic styles when experimental DS is on

## Short description
This PR allows font scaling to all typographic styles. This change allows virtually all dynamic sizing on (almost) every DS component.

> [!important]
> For testing purposes, **font scaling currently only works when experimental DS is enabled**.

## List of changes proposed in this pull request
- Add `allowFontScaling` to all the typographic styles
- Add the new `maxFontSizeMultiplier` constant to `IOVisualConstants`
- Dynamically trim text box in the `Alert` component, based on dynamic font scale. Manual fixing has been introduced by this PR:
  - https://github.com/pagopa/io-app-design-system/pull/282
- **[extra]** Fix some visual issues of the `Module…` and `ListItem…` components

### Preview

https://github.com/pagopa/io-app-design-system/assets/1255491/20b3216a-b37c-4da3-9f52-68b72ca1af04



## How to test
1. Launch the example app
2. Change font zoom from the OS
3. Check all the changes applied to the components
